confignode/src/main/java/org/apache/iotdb/confignode/client/AsyncDataNodeClientPool.java

```diff
@@ -58,13 +57,106 @@ public class AsyncDataNodeClientPool {
 
   private final IClientManager<TEndPoint, AsyncDataNodeInternalServiceClient> clientManager;
 
+  private final int retryNum = 6;
+
   private AsyncDataNodeClientPool() {
     clientManager =
         new IClientManager.Factory<TEndPoint, AsyncDataNodeInternalServiceClient>()
             .createClientManager(
                 new ConfigNodeClientPoolFactory.AsyncDataNodeInternalServiceClientPoolFactory());
   }
 
+  /**
+   * When the request fails, reconnect the DataNode that failed the request
+   *
+   * @param req request
+   * @param handlerMap Map<index, Handler>
+   * @param dataNodeLocations ConcurrentHashMap<index, TDataNodeLocation> The specific DataNodes
+   */
+  public void sendAsyncRequestToDataNodeWithRetry(
+      Object req,
+      Object handlerMap,
+      ConcurrentHashMap<Integer, TDataNodeLocation> dataNodeLocations) {
```

Comment:
```suggestion
      List<TDataNodeLocation> dataNodeLocations) {
```
Which is enough for sending asynchronize requests

Reply:
Because createRegions is special, there may be multiple requests for the same node, so index is used instead of datanodeid as a key.And used Map<index, TdataNodeLocation>